lib/ecmascript.ts

```diff
@@ -4591,10 +4606,16 @@ export function DifferenceTemporalPlainYearMonth(
   const roundingIncrement = ToTemporalRoundingIncrement(options, undefined, false);
 
   const fieldNames = CalendarFields(calendar, ['monthCode', 'year'] as const);
-  const otherFields = ToTemporalYearMonthFields(other, fieldNames);
-  const thisFields = ToTemporalYearMonthFields(yearMonth, fieldNames);
-  const otherDate = CalendarDateFromFields(calendar, { ...otherFields, day: 1 });
-  const thisDate = CalendarDateFromFields(calendar, { ...thisFields, day: 1 });
+  const otherFields = ToTemporalYearMonthFields(other, fieldNames) as Required<Temporal.PlainYearMonthLike> & {
+    day: number;
+  };
+  otherFields.day = 1;
+  const thisFields = ToTemporalYearMonthFields(yearMonth, fieldNames) as Required<Temporal.PlainYearMonthLike> & {
```

Comment:
A few questions about this cast:
* The return type of `ToTemporalYearMonthFields` is `Required<PrimitiveFieldsOf<Temporal.PlainYearMonthLike>>`, not `Required<Temporal.PlainYearMonthLike>`. To avoid mismatches like this, I will often use the return type directly, e.g. `ReturnType<ToTemporalYearMonthFields> & { day: 1 }`.
* If we use this cast twice, should we create a local type declaration and re-use it?

Reply:
Great points - done. Also added an explanation for why this is even needed.